core/src/main/java/com/homeaway/streamplatform/streamregistry/service/impl/StreamServiceImpl.java

```
@@ -0,0 +1,247 @@
+            topicConfig.putAll(stream.getTopicConfig());
+        }
+
+        String bootstrapServer = clusterValue.getClusterProperties().get(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG);
+        String zkConnect = clusterValue.getClusterProperties().get(KafkaProducerConfig.ZOOKEEPER_QUORUM);
+        topicConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServer);
+        topicConfig.put(KafkaProducerConfig.ZOOKEEPER_QUORUM, zkConnect);
+
+        kafkaManager.upsertTopics(Collections.singleton(stream.getName()), stream.getPartitions(), stream.getReplicationFactor(), topicConfig, isNewStream);
+        log.info("Topic {} created/updated at {}", stream.getName(), bootstrapServer);
+    }
+
+    private void applyDefaultHint(Stream stream) {
+        Tags tags = stream.getTags();
+
+        String hint = tags.getHint();
+
+        if (hint == null || hint.trim().matches("(?i:string)?")) {
+            hint = AbstractService.PRIMARY_HINT;
+        }
+
+        tags.setHint(hint);
+    }
+
+    @Override
+    public Stream getStream(String streamName) throws StreamNotFoundException {
+        log.info("Pulling stream information from global state-store for streamName={}", streamName);
+        Optional<AvroStream> streamValue = streamDao.getStream(streamName).getValue();
+
+        if (!streamValue.isPresent()) {
+            throw new StreamNotFoundException(String.format("Stream=%s not found.", streamName));
+        } else {
+            return AvroToJsonDTO.convertAvroToJson(streamValue.get());
+        }
+    }
+
+    @Override
+    public void deleteStream(String streamName) throws StreamNotFoundException{
+        Pair<AvroStreamKey, Optional<AvroStream>> keyValue = streamDao.getStream(streamName);
+
+        if (!keyValue.getValue().isPresent()) {
+            throw new StreamNotFoundException(String.format("Stream=%s not found.", streamName));
+        }
+        try {
+            streamDao.upsertStream(keyValue.getKey(), null);
```

Do you really want the service to swallow all exceptions ? Shouldn't runtime exceptions be bubbled up to the resource layer to be handled as 500s?

---

When i refactored i missed it. 
Fixed it now. Throwing back all the Runtime Exceptions and Resource class is equipped to handle it properly